pkg/controllermanager/controller/shoot/shoot_control.go

```diff
@@ -200,160 +149,163 @@ func (c *Controller) updateShootStatusProcessing(shoot *gardenv1beta1.Shoot, mes
-	key, err := cache.MetaNamespaceKeyFunc(shootObj)
-	if err != nil {
-		return true, err
+	if shoot.Spec.Cloud.Seed != nil {
+		if err := c.runDeleteShootFlow(o); err != nil {
+			o.Logger.Errorf("Error during execution of the delete operation for shoot: %+v", err.Description)
+			c.recorder.Event(shoot, corev1.EventTypeWarning, gardenv1beta1.EventDeleteError, err.Description)
+			if err := c.updateShootStatusDeleteError(o, err); err != nil {
+				o.Logger.WithError(err).Error("Could not update the Shoot status after deletion error")
+				return reconcile.Result{}, err
+			}
+			return reconcile.Result{}, errors.New(err.Description)
+		}
+		if err := o.DeleteClusterResourceFromSeed(context.TODO()); err != nil {
+			lastErr := &gardencorev1alpha1.LastError{Description: fmt.Sprintf("Could not delete Cluster resource in seed: %s", err)}
+			c.recorder.Event(shoot, corev1.EventTypeWarning, gardenv1beta1.EventDeleteError, lastErr.Description)
+			if err := c.updateShootStatusDeleteError(o, lastErr); err != nil {
+				o.Logger.WithError(err).Errorf("Could not update the Shoot status after deletion error")
+				return reconcile.Result{}, err
+			}
+			return reconcile.Result{}, errors.New(lastErr.Description)
+		}
 	}
-	operationID, err := utils.GenerateRandomString(8)
-	if err != nil {
-		return true, err
+
+	c.recorder.Event(shoot, corev1.EventTypeNormal, gardenv1beta1.EventDeleted, "Deleted Shoot cluster")
+	if err := c.updateShootStatusDeleteSuccess(o); err != nil {
+		o.Logger.WithError(err).Errorf("Could not update the Shoot status after deletion success")
+		return reconcile.Result{}, err
 	}
+	return reconcile.Result{}, nil
+}
 
+func (c *Controller) reconcileShoot(shoot *gardenv1beta1.Shoot, o *operation.Operation) (reconcile.Result, error) {
 	var (
-		shoot         = shootObj.DeepCopy()
-		shootLogger   = logger.NewShootLogger(logger.Logger, shoot.Name, shoot.Namespace, operationID)
-		operationType = gardencorev1alpha1helper.ComputeOperationType(shoot.ObjectMeta, shoot.Status.LastOperation)
+		operationType   = gardencorev1alpha1helper.ComputeOperationType(shoot.ObjectMeta, shoot.Status.LastOperation)
+		failedOrIgnored = common.IsShootFailed(shoot) || common.ShouldIgnoreShoot(c.respectSyncPeriodOverwrite(), shoot)
 	)
 
-	logger.Logger.Infof("[SHOOT RECONCILE] %s", key)
-	shootJSON, _ := json.Marshal(shoot)
-	shootLogger.Debugf(string(shootJSON))
-
-	// We check whether the Shoot's last operation status field indicates that the last operation failed (i.e. the operation
-	// will not be retried unless the shoot generation changes).
-	if shootIsFailed(shoot) {
-		if shoot.Status.Gardener.Version == version.Get().GitVersion {
-			shootLogger.Infof("Will not reconcile as the last operation has been set to '%s' and the generation has not changed since then.", gardencorev1alpha1.LastOperationStateFailed)
-			return false, nil
+	if shoot.Spec.Cloud.Seed != nil {
+		if err := c.checkShootIsOnHealthySeed(shoot); err != nil {
```

Comment:
On a new gardener Identity I noticed a lot of log messages when the Shoot is not on a healthy seed. 

I guess this is because all the goroutines/workers reconcile concurrently  -> update the condition with new timestamp -> trigger an update -> further reconciles.     
It also takes some time in my observation until the seeds are reconciled to stop the flooding of the logs.

We might consider a RateLimitedQueue like [here ](https://github.com/gardener/gardener/blob/master/pkg/scheduler/controller/scheduler.go#L70)

Reply:
We already use them, see
https://github.com/gardener/gardener/blob/master/pkg/controllermanager/controller/shoot/shoot.go#L79
and
https://github.com/gardener/gardener/blob/master/pkg/controllermanager/controller/shoot/shoot.go#L83
Since it's exponential backoff, at first you'll have more logs and later on fewer.